### DISCUSSION TITLE
Is there any option for tree shaking from an object with a dynamic string?

### DISCUSSION THREAD

**Discussion by jnlasher** [webpack/webpack#18100] [Q&A]
I have a library file that exports a lot of components as a top level object, e.g.
```
const A = {};
const B = {};
const C = {};
// ... so on
const Library = { A, B, C, /* ... and so on ... /* };
```

This is being dereferenced from another file based on the name:
```
import { Library } from './Library';
function resolve(path) {
  Library[path];
}
```

It makes sense that webpack can't analyze this since the imports are so dynamic. I'm kind of wondering if there are any options to make this more tree-shakable or I have to just rewrite this to export each object individually?

**Comment by alexander-akait** [comment]
Hello, we have this https://webpack.js.org/api/module-methods/#magic-comments, i.e. you can use `webpackExports`
